dissect/target/plugins/general/plugins.py

```diff
@@ -94,15 +101,128 @@ def get_description_dict(
     return output_descriptions
 
 
+@dataclass
+class PluginFunction:
+    name: str
+    output_type: str
+    class_object: str
+    method_name: str
+    plugin_desc: dict
+
+
 class PluginListPlugin(Plugin):
     def check_compatible(self):
         return True
 
-    @export(output="none")
-    @arg("--docs", dest="print_docs", action="store_true")
-    def plugins(self, print_docs=False):
-
-        categorized_plugins = dict(sorted(categorize_plugins().items()))
+    @export(output="none", cache=False)
+    def plugin_function_db(self) -> Tuple[dict, list]:
+        db = []
+        rootset = set()
+
+        def all_plugins():
+            yield from plugin.plugins()
+            yield from plugin.os_plugins()
+            yield from plugin.child_plugins()  # Although nothing is exported here...
+
+        for available in all_plugins():
+            # filter other-os plugins
+            if self.target._os_plugin is not None:
+                if (
+                    available["module"].endswith("_os")
+                    and self.target._os_plugin.__module__.find(available["module"]) == -1
+                ):
+                    continue
+            rootset.add(available["module"].split(".")[0])
+            for exported in available["exports"]:
+                if exported == "get_all_records":
+                    continue
+                entry = {
+                    "name": f"{available['module']}.{exported}",
+                    "plugin": available,
+                }
+                db.append(entry)
+
+        return db, rootset
+
+    @export(output="none", cache=False)
+    def find_plugin_functions(
+        self, patterns: str, children: bool = False, compatibility: bool = False
+    ) -> List[PluginFunction]:
+        result = []
+        functions, rootset = self.plugin_function_db()
+
+        for pattern in patterns.split(","):
+
+            wildcard = any(char in pattern for char in ["*", "!", "?", "[", "]"])
+            treematch = pattern.split(".")[0] in rootset and pattern != "os"
+
+            if treematch and not wildcard:
+                pattern += "*"  # this seems to be what people expect...
```

Comment:
Shouldn't this be `.*` as the pattern ends in the last tree part, e.g. `foo.bar` -> `foo.bar.*` instead of `foo.bar*`?